### DISCUSSION TITLE
Etcd nodes IO rate got higher and snapshot size got bigger

### DISCUSSION THREAD

**Discussion by serhiynovos** [rancher/rke2#5680] [Q&A] [answered]
I just noticed that IO is significantly increased on my RKE2 cluster with 3 servers and 3 agents. I checked I did not do any new significant workload deployments this week except upgrading Rancher from 2.8.2 to 2.8.3 and then upgrading this cluster from 1.27.12 version to v1.28.8 +rke2r1. As I remember I was doing it about 1 week ago. Maybe somebody knows what exactly reason can be which will be very helpful
<img width="1283" alt="Знімок екрана 2024-04-05 о 12 58 29" src="https://github.com/rancher/rke2/assets/9201885/f9a728b9-dd4c-4d65-9d99-e243df9228e7">
<img width="397" alt="Знімок екрана 2024-04-05 о 13 02 58" src="https://github.com/rancher/rke2/assets/9201885/1691ca82-bcd2-4cd1-9c24-65133a7ed56b">

**Comment by brandond** [comment]
IO rate corresponds to create/modify/delete load from the apiserver. Snapshot size corresponds to objects in the datastore. It sounds like something caused an increase in both traffic and the number of resources. The apiserver exposes request rate and resource count metrics, I would recommend looking at these to try to determine what's changed.

**Reply by serhiynovos** [reply]
Is there any reasons why it can be increased after k8s update ?

**Reply by brandond** [reply]
That is not a simple question to answer. I can say that I would not **expect** it to change when only changing the kubernetes version. You are really going to need to look at metrics to determine what's happening in your cluster though.

**Reply by serhiynovos** [reply]
Is there any documents describing what exactly I can look what is causing this issue ?

**Reply by brandond** [reply]
https://kubernetes.io/docs/reference/instrumentation/metrics/
I would probably start with:
* apiserver_storage_objects
* apiserver_request_total

**Reply by serhiynovos** [reply]
I install rancher-monitoring, can I see there these values ?

**Reply by serhiynovos** [reply]
<img width="283" alt="Знімок екрана 2024-04-05 о 22 09 12" src="https://github.com/rancher/rke2/assets/9201885/fa26fde1-2c11-47d9-98ea-8637d1b133ff">

I see longhorn is doing a lot of requests

**Reply by serhiynovos** [reply]
<img width="408" alt="Знімок екрана 2024-04-05 о 22 13 14" src="https://github.com/rancher/rke2/assets/9201885/f3f83d9a-ddf5-4a8a-895c-5228fe91749b">

I see it has  a lot of events

**Reply by brandond** [reply]
ok, so go do `kubectl get event -A` on that cluster and see what is being noisy.

**Reply by serhiynovos** [reply]
Yes it shows bunch of them. So I guess I can just go and delete them all. Right ? `kubectl delete events --all -A`

**Reply by brandond** [reply]
If you haven't identified what is creating them, and taken steps to address it, it is likely they will just come back again. Events automatically expire and are deleted after 1 hour, so if you have a lot of them, it means they are being created continuously.

Solve the problem, not the symptom.

**Reply by serhiynovos** [reply]
Is it fine that longhorn is making so many GET and PUT api calls ?

**Reply by serhiynovos** [reply]
<img width="1407" alt="Знімок екрана 2024-04-05 о 22 48 10" src="https://github.com/rancher/rke2/assets/9201885/745a304b-b632-404a-92b0-856d7deb8e1e">

I see some of services are using cpu constantly

**Reply by serhiynovos** [reply]
<img width="1258" alt="Знімок екрана 2024-04-05 о 22 55 25" src="https://github.com/rancher/rke2/assets/9201885/cac3aaa4-6b63-4fa5-9962-4a1b7825f7d1">

And I see apiserver node always under the load

**Reply by brandond** [reply]
I don't see any info on what the events are from. Is it LH?

**Reply by serhiynovos** [reply, marked as answer]
Yes. A lot of put and get requests. Looks like it's related to this issue 

https://github.com/longhorn/longhorn/issues/8114
